controllers/rate_limiting_limits_controller.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	if limitadorCR.GetNamespace() != kuadrantList.Items[0].Namespace {
+		logger.Info("this limitador resource is not managed by kuadrant, unexpected namespace")
+		return ctrl.Result{}, nil
+	}
+
+	rlps, err := r.readRLPs(ctx, limitadorCR.GetNamespace())
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	rateLimitIndex := rlptools.NewRateLimitIndex()
+
+	for _, rlp := range rlps {
+		if _, ok := rateLimitIndex.Get(client.ObjectKeyFromObject(rlp)); ok {
+			continue
+		}
+
+		rateLimitIndex.Set(client.ObjectKeyFromObject(rlp), rlptools.LimitadorRateLimitsFromRLP(rlp))
+	}
+
+	// return if limitador is up to date
+	if rlptools.Equal(rateLimitIndex.ToRateLimits(), limitadorCR.Spec.Limits) {
+		logger.V(1).Info("limitador is up to date, skipping update")
+		return ctrl.Result{}, nil
+	}
+
+	// update limitador
+	limitadorCR.Spec.Limits = rateLimitIndex.ToRateLimits()
+	err = r.UpdateResource(ctx, limitadorCR)
+	logger.V(1).Info("update limitador", "limitador", client.ObjectKeyFromObject(limitadorCR), "err", err)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Rate limiting limits reconciled successfully")
+	return ctrl.Result{}, nil
+}
+
+// Rate limit policies targeting programmed gateways or routes accepted by parent gateways.
+func (r *RateLimitingLimitsReconciler) readRLPs(ctx context.Context, kuadrantNS string) ([]*kuadrantv1beta2.RateLimitPolicy, error) {
+	logger, err := logr.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// When multiple kuadrant instances are supported, this fetching could be targeted to all
+	// gateways for the same kuadrant instance of the reconciled limitador instance
+	gwList := &gatewayapiv1.GatewayList{}
+	// Get all the routes having the gateway as parent
+	err = r.Client().List(ctx, gwList)
+	logger.V(1).Info("topology: list gateways", "#Gateways", len(gwList.Items), "err", err)
+	if err != nil {
+		return nil, err
+	}
+
+	routeList := &gatewayapiv1.HTTPRouteList{}
+	// Get all the routes having the gateway as parent
+	err = r.Client().List(ctx, routeList)
+	logger.V(1).Info("topology: list httproutes", "#HTTPRoutes", len(routeList.Items), "err", err)
+	if err != nil {
+		return nil, err
+	}
+
+	rlpList := &kuadrantv1beta2.RateLimitPolicyList{}
+	// Get all the rate limit policies
+	err = r.Client().List(ctx, rlpList)
+	logger.V(1).Info("topology: list rate limit policies", "#RLPS", len(rlpList.Items), "err", err)
+	if err != nil {
+		return nil, err
+	}
+
+	policies := utils.Map(rlpList.Items, func(p kuadrantv1beta2.RateLimitPolicy) kuadrantgatewayapi.Policy { return &p })
+
+	t, err := kuadrantgatewayapi.NewTopology(
+		kuadrantgatewayapi.WithGateways(utils.Map(gwList.Items, ptr.To)),
+		kuadrantgatewayapi.WithRoutes(utils.Map(routeList.Items, ptr.To)),
+		kuadrantgatewayapi.WithPolicies(policies),
+		kuadrantgatewayapi.WithLogger(logger),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	topologyindexes := kuadrantgatewayapi.NewTopologyIndexes(t)
```

Comment:
TODO: Common function to load the topology for all routes, gateways and policies.